libbeat/processors/actions/truncate_fields.go

```diff
@@ -0,0 +1,140 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package actions
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+type truncateFieldsConfig struct {
+	Fields        []string `config:"fields"`
+	MaxBytes      int      `config:"max_bytes" validate:"min=0,nonzero"`
```

Comment:
Is this to be used for working with `ignore_above`? I’m pretty sure `ignore_above` is based on utf-8 character count (I think [this](
https://github.com/elastic/elasticsearch/blob/67b5a83a9ac1e28141ceee390e4aabe601192b14/server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java#L345) is the check). So an option that counts/truncates base on runes would be nice.

Reply:
I am following the same algorithm in `readfile.LimitReader` to have the same result as the user would expect from using `max_bytes` on the input level configuration.

Do you mean adding a new option to the processor which sets whether the truncation should be done based on runes or bytes?

Reply:
Firstly I'm sure how this is going to be used so I'm not sure how important this is. But yes, if it's for helping keep the field length at the `ignore_above` limit then I was thinking of a separate option that's mutually exclusive with `max_bytes`, like `max_characters`. This option would count UTF-8 characters such that the units are consistent with `ignore_above`.

So if you are ingesting lots of internationalized logs with non-ascii characters this will get you closer to the `ignore_above` limit than `max_bytes` would.

But I will leave it someone else to decide how important such a feature is.

Reply:
Personally, I like the idea.
@ruflin WDYT?

Reply:
It would be nice to work with runes, I like the proposal of @andrewkroh

Reply:
+1